elastic/apm-server
# Dockerfile: define the apm-server user

## Motivation/summary

In the Dockerfile refactoring for 8.5, I made a change to use `USER 1000` rather than `USER apm-server`,
obviating the need to install useradd/adduser in ubi8. However, this breaks users who expect the apm-server user to exit.

Instead of installing useradd/adduser, create the /etc/passwd entry manually.

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. `make package-docker`
2. `docker run --rm $(cat build/docker/apm-server-8.6.0.txt) whoami` should print "apm-server"
3. `docker run --rm $(cat build/docker/apm-server-ubi8-8.6.0.txt) whoami` should print "apm-server"

## Related issues

Closes #9495
